### PR TITLE
[Feature] 🔨 Implement all comparison operations 

### PR DESCRIPTION
**Pull request by posidoni**
## Implementation of comparisons functions:

**UT status:**
- passes all tests

image

**✅ GCOV coverage:**
<br>
image

### CODE REVIEW

File: src/binary_api/s21_core.h [hunk 1]
```diff
@@ -1,10 +1,10 @@
-#include "../s21_decimal.h"
```
Comment (posidoni) [inline src/binary_api/s21_core.h:1]: Clang marked recursive includes as error. This include was an obvious error, I don't know why it wasn't fixed earlier. This fix required bucnch of other fixes in includes structure, which you can see throughout this PR.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Session.vim
 notes.md
 test
+.vscode/settings.json
```
Comment (posidoni) [inline .gitignore]: @bezlant interresting syntax, isn't it? Make notes :)

File: src/Makefile [hunk 3]
```diff
@@ -2,7 +2,7 @@
 CC 				 =	gcc
 
 # Compilation flags configuration
-ALL_TARGETS 	 = $(TARGET) test gcov_report 
+ALL_TARGETS 	 = $(TARGET) test # gcov_report
```
Comment (posidoni) [inline src/Makefile]: Nothing has changed. Sorry for committing whitespace changes. I will try to eliminate this kind of commits.

File: src/tests/test_runner.c [hunk 4]
```diff
@@ -32,27 +32,27 @@ void run_testcase(Suite *testcase) {
...
-                           //    suite_s21_from_decimal_to_float(),
-                           //    suite_s21_floor(),
-                           //    suite_s21_round(),
-                           //    suite_s21_truncate(),
-                           //    suite_s21_negate(),
-                           NULL};
+        // suite_s21_add(),
+        //   suite_s21_sub(),
+        //    suite_s21_mul(),
+        // suite_s21_div(),
```
Comment (posidoni) [inline src/tests/test_runner.c:38]: Functions which are uncommented are stable. Yay! ^_^

File: src/binary_api/s21_exponent.c [hunk 5]
```diff
@@ -1,19 +1,34 @@
...
+ * exponent. This is done via multiplication by 10 of decimal with the
+ * smallest exponent.
+ *
+ * In case of overflow the flag is raised.
+ *
+ * @param [a, b] - pair of decimals that need to be normalized to the same
+ * exponent
+ */
+
+void s21_normalize_decimal_pair(s21_decimal *a, s21_decimal *b, int *overflow) {
```
Comment (posidoni) [inline src/binary_api/s21_exponent.c:14]: This is very important function for is_less / is_greater that normalizes pair of decimals. Normalize = make their exponent equal, so we can compare them.

File: src/binary_api/s21_core.c [hunk 6]
```diff
@@ -1,8 +1,4 @@
-#include "s21_core.h"
-
-void init_decimal(s21_decimal *decimal) {
-    memset(decimal, 0, sizeof(*decimal));
```
Comment (posidoni) [inline src/binary_api/s21_core.c:4]: Removed deprecated fn.

File: src/conversion/s21_from_int_to_decimal.c [hunk 7]
```diff
@@ -1,9 +1,10 @@
+
 #include "../s21_decimal.h"
 
 int s21_from_int_to_decimal(int src, s21_decimal *dst) {
     convertation_result status = CONVERTATION_OK;
     if (dst) {
-        init_decimal(dst);
```
Comment (posidoni) [inline src/conversion/s21_from_int_to_decimal.c:6]: Removed deprecated function call. This change is stable, passes all tests.

File: src/tests/comparison_tests/test_s21_is_equal_not_equal.c [hunk 8]
```diff
@@ -96,9 +96,9 @@ START_TEST(equality_all_zeros) {
 END_TEST
 
 START_TEST(hardcoded_decimal_loop) {
-    s21_decimal n1 = get_random_decimal_size(get_rand(0, 96), 0, get_rand(0, 26));
```
Comment (posidoni) [inline src/tests/comparison_tests/test_s21_is_equal_not_equal.c:99]: Removed deprecated function calls.

File: src/tests/conversion_tests/decimal_conversion/test_s21_from_int_to_decimal.c [hunk 9]
```diff
@@ -9,71 +9,67 @@ START_TEST(NULL_POINTER) {
 
 START_TEST(ZERO_INT) {
     int test_int = 0;
-    s21_decimal *decimal = malloc(sizeof(s21_decimal));
```
Comment (posidoni) [inline src/tests/conversion_tests/decimal_conversion/test_s21_from_int_to_decimal.c:12]: Removed unnesessary mallocs. These tests do work without them.

File: src/tests/helpers/random_values.c [hunk 10]
```diff
@@ -1,7 +1,11 @@
 #include "../s21_decimal_test.h"
 
 static void set_random_pair_sign(s21_decimal *dec, mpz_t *in_mpz_copy);
+
+#ifdef DEBUG
```
Comment (posidoni) [inline src/tests/helpers/random_values.c:5]: Currently ALL debug-like prints are guarded in our project.

File: src/tests/s21_decimal_test.h [hunk 11]
```diff
@@ -29,11 +28,10 @@
 void apply_exponent_to_decimal(s21_decimal *dec);
 
 // decimal manipulation
-s21_decimal get_random_int_decimal(void);
```
Comment (posidoni) [inline src/tests/s21_decimal_test.h:32]: Removed deprecated functions signatures.

File: .gitignore [hunk 12]
```diff
@@ -1,6 +1,8 @@
 Session.vim
 notes.md
 test
+.vscode/*
+!.vscode/launch.json
```
Comment (posidoni) [inline .gitignore:5]: Interesting syntax. Isn't it? :) Make notes @bezlant

File: src/comparison/s21_is_less_or_equal.c [hunk 13]
```diff
@@ -1,5 +1,5 @@
 #include "../s21_decimal.h"
 
 int s21_is_less_or_equal(s21_decimal a, s21_decimal b) {
-    return s21_is_equal(a, b) || s21_is_less(a, b);
+    return s21_is_equal(a, b) || s21_is_less(a, b );
```
Comment (bezlant) [inline src/comparison/s21_is_less_or_equal.c:4]: This is redundant 🗒️

**Review by bezlant — COMMENTED** [review]
Thanks for the work done. This is very crucial for our project.

**Pull request merged**